src/claim_canister/src/lib.rs

```diff
@@ -53,8 +80,77 @@ pub fn add_claim(
     Ok(claim_id)
 }
 
+async fn transfer_from_caller(from: Principal, amount: Nat) -> Result<(), ClaimError> {
```

Comment:
Support icrc1 standard instead icrc2
you should have only one helper function for transfers like in pool canister.